packages/app-store/googlecalendar/lib/CalendarService.test.ts

```diff
@@ -45,53 +42,160 @@ const testSelectedCalendar = {
+
+  // Create cache
+  await prismock.calendarCache.create({
+    data: {
+      credentialId: credentialInDb1.id,
+      key: JSON.stringify({
+        timeMin: getTimeMin(dateFrom1),
+        timeMax: getTimeMax(dateTo1),
+        items: [{ id: testSelectedCalendar.externalId }],
+      }),
+      value: {
+        calendars: [
+          {
+            busy: [
+              {
+                start: "2023-12-01T18:00:00Z",
+                end: "2023-12-01T19:00:00Z",
+              },
+            ],
+          },
+        ],
+      },
+      expiresAt: String(Date.now() + 10000),
     },
-    "xxxxxxxxxxxxxxxxxxxxxxxxxx@group.calendar.google.com": { busy: [] },
-  },
-};
+  });
 
-const calendarCacheResponse = {
-  key: "dummy",
-  expiresAt: new Date(),
-  credentialId: 1,
-  value: testFreeBusyResponse,
-};
+  oAuthManagerMock.OAuthManager = defaultMockOAuthManager;
+  const calendarService = new CalendarService(credentialInDb1);
+
+  // Test cache hit
+  const data = await calendarService.getAvailability(dateFrom1, dateTo1, [testSelectedCalendar]);
+  expect(data).toEqual([
+    {
+      start: "2023-12-01T18:00:00Z",
+      end: "2023-12-01T19:00:00Z",
+    },
+  ]);
 
-test("Calendar Cache is being called", async () => {
-  prismaMock.calendarCache.findUnique
-    // First call won't have a cache
-    .mockResolvedValueOnce(null)
-    // Second call will have a cache
-    .mockResolvedValueOnce(calendarCacheResponse);
-
-  // prismaMock.calendarCache.create.mock.
-  const calendarService = new CalendarService(testCredential);
-  vi.spyOn(calendarService, "authedCalendar").mockReturnValue({
-    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
-    //@ts-ignore - Mocking the authedCalendar so can't return the actual response
-    freebusy: {
-      query: vi.fn().mockReturnValue({
-        data: testFreeBusyResponse,
+  const credentialInDb2 = await createCredentialInDb();
+  const dateFrom2 = new Date(Date.now()).toISOString();
+  // Tweak date so that it's a cache miss
+  const dateTo2 = new Date(Date.now() + 100000000).toISOString();
+  const calendarService2 = new CalendarService(credentialInDb2);
+
+  // Test Cache Miss
+  await calendarService2.getAvailability(dateFrom2, dateTo2, [testSelectedCalendar]);
+
+  // Expect cache to be updated in case of a MISS
+  const calendarCache = await prismock.calendarCache.findFirst({
+    where: {
+      credentialId: credentialInDb2.id,
+      key: JSON.stringify({
+        timeMin: getTimeMin(dateFrom2),
+        timeMax: getTimeMax(dateTo2),
+        items: [{ id: testSelectedCalendar.externalId }],
       }),
     },
   });
 
-  await calendarService.getAvailability(new Date().toISOString(), new Date().toISOString(), [
-    testSelectedCalendar,
-  ]);
-  await calendarService.getAvailability(new Date().toISOString(), new Date().toISOString(), [
-    testSelectedCalendar,
-  ]);
-  expect(prismaMock.calendarCache.findUnique).toHaveBeenCalled();
```

Comment:
We can now actually test that the cached value is returned and not just that the function is called.